app/models/fluidigm_template_task.rb

```diff
@@ -6,7 +6,8 @@ def partial
   def plate_purpose_options(batch) # rubocop:todo Metrics/AbcSize
     requests = batch.requests.flat_map(&:next_requests)
     plate_purposes = requests.filter_map(&:request_type).uniq.map(&:acceptable_plate_purposes).flatten.uniq
-    plate_purposes = batch.requests.filter_map { |r| r.request_metadata.target_purpose }.uniq if plate_purposes.empty? # Fallback situation for the moment
+    plate_purposes = # Fallback situation for the moment
+      batch.requests.filter_map { |r| r.request_metadata.target_purpose }
```

Comment:
This doesn't look like it's functionally the same.  There's no guard to prevent overwriting `plate_purposes` if it is not `empty?` and the `.uniq` has been dropped.

Reply:
What the flip happened there?! I rolled it back, then ran rubocop over it again, only for it to spit out this beauty:
```ruby
class FluidigmTemplateTask < PlateTemplateTask # rubocop:todo Style/Documentation
  def partial
    'fluidigm_template_batches'
  end

  def plate_purpose_options(batch) # rubocop:todo Metrics/AbcSize
    requests = batch.requests.flat_map(&:next_requests)
    plate_purposes = requests.filter_map(&:request_type).uniq.map(&:acceptable_plate_purposes).flatten.uniq
    plate_purposes = # Fallback situation for the moment
batch.requests.filter_map do |r|
 r.request_metadata.target_purpose end.uniq
    plate_purposes.map { |p| [p.name, p.size, p.id] }.sort
  end
end
```

Which is also not the same. Prettier then reformatted it to something more akin to what was there, but with the uniq at least.